src/modules/esl-event-listener/core/targets/resize.adapter.ts

```diff
@@ -1,47 +1,54 @@
 import {SyntheticEventTarget} from '../../../esl-utils/dom/events/target';
 import {overrideEvent} from '../../../esl-utils/dom/events/misc';
 
+import type {ESLDomElementTarget} from '../../../esl-utils/abstract/dom-target';
+
+export type ESLResizeTarget = ESLDomElementTarget | Window;
+
 /** Adapter class for {@link ResizeObserver} that implements {@link EventTarget} */
 export class ESLResizeObserverTarget extends SyntheticEventTarget {
   /** {@link ESLResizeObserverTarget} instances holder */
-  protected static readonly mapping = new WeakMap<Element, ESLResizeObserverTarget>();
+  protected static readonly mapping = new WeakMap<Element | EventTarget, ESLResizeObserverTarget>();
   /** {@link ResizeObserver} instance to observe DOM element related to {@link ESLResizeObserverTarget} */
   protected static readonly observer$$ = new ResizeObserver((changes: ResizeObserverEntry[]) =>
     changes.forEach(this.handleChange, this)
   );
 
+  /** Observed {@link ESLResizeTarget} of the {@link ESLResizeObserverTarget} instance */
+  public readonly target: Element | Window;
+
   /** Internal method to handle {@link ResizeObserver} entry change */
   protected static handleChange(
     this: typeof ESLResizeObserverTarget,
-    detail: ResizeObserverEntry
+    detail: ResizeObserverEntry | Event
   ): void {
+    if (!detail.target) return;
     const adapter = this.mapping.get(detail.target);
     if (!adapter) return;
     const event = new CustomEvent('resize', {detail});
     overrideEvent(event, 'target', adapter.target);
     adapter.dispatchEvent(event);
   }
 
-  /** Creates {@link ESLResizeObserverTarget} instance for the {@link Element} */
-  public static create(target: Element): ESLResizeObserverTarget {
+  /** Creates {@link ESLResizeObserverTarget} instance for the {@link ESLResizeTarget} */
+  public static create(target: ESLResizeTarget): ESLResizeObserverTarget {
     return new ESLResizeObserverTarget(target);
   }
 
   /**
-   * Creates {@link ESLResizeObserverTarget} for the {@link Element}.
-   * Note the {@link ESLResizeObserverTarget} instances are singletons relatively to the {@link Element}
+   * Creates {@link ESLResizeObserverTarget} for the {@link ESLResizeTarget}.
+   * Note the {@link ESLResizeObserverTarget} instances are singletons relatively to the {@link ESLResizeTarget}
    */
-  protected constructor(
-    /** Observed {@link Element} of the {@link ESLResizeObserverTarget} instance */
-    public readonly target: Element
-  ) {
+  protected constructor(target: ESLResizeTarget) {
+    if (!(target instanceof Window || target instanceof Element)) target = target.$host;
```

Comment:
Consider mowing to esl-utils/abstract/dom-target as utility `resolveDomTarget`
That can also be applied to other cases of ESLDomElementTarget usage